src/async_fuse/memfs/dist/server.rs

```diff
@@ -201,18 +201,18 @@ async fn load_dir<S: S3BackEnd + Send + Sync + 'static, K: KVEngine + 'static>(
     meta: Arc<S3MetaData<S, K>>,
     path: &str,
 ) -> anyhow::Result<()> {
-    let inum_opt = {
-        let path2inum = meta.path2inum.read().await;
-        path2inum.get(path).copied()
-    };
+    let inum_opt = { meta.get_inum_from_kv_engine(path).await };
 
     match inum_opt {
         None => tcp::write_message(stream, response::load_dir_none().as_slice()).await?,
-        Some(inum) => match meta.cache.read().await.get(&inum) {
+        Some(inum) => match meta.get_node_from_kv_engine(inum).await.as_ref() {
             None => tcp::write_message(stream, response::load_dir_none().as_slice()).await?,
             Some(node) => {
-                tcp::write_message(stream, response::load_dir(node.get_dir_data()).as_slice())
-                    .await?
+                tcp::write_message(
+                    stream,
+                    response::load_dir(node.as_ref().get_dir_data()).as_slice(),
+                )
```

Comment:
I think we use tcp to communicate with other nodes for metadata updating. Do we still need it in current design?